### DISCUSSION TITLE
Trying to replicate custom itermocil layout in Kitty

### DISCUSSION THREAD

**Discussion by freak4pc** [kovidgoyal/kitty#8062] [Q&A] [answered]
Hey all :) 
In our company we use itermocil which creates a grid-like layout with 6 internal windows, each running its own command. 
I'm trying to replicate this same iTerm-specific layout in Kitty windows.

This is what I have so far:

```
new_os_window
layout grid
cd ...
launch --title "Webpack" nvm use && npm install && npm run start-webpack
[...5 more launch statements that work correctly]
```

I run it using
```
kitty --session ~/.config/kitty/my-custom-layout.yml
```

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/bfd9fde8-1827-430b-a8a4-522f8be99e23">

I tried to replicate it but have a few issues:

1. That `nvm` command doesn't work, that single window writes:
```
Failed to launch child: nvm
With error: No such file or directory
Press Enter or Esc to exit
```
I imagine it doesn't have any environment because of the way it executes, but I wondered if there's any way to fix it. Specifically let it run ~/.zshrc or other environment-enabling scripts. I tried doing `source ~/.zshrc` in my launch but that doesn't work either (same error).

2. I would much prefer my command to create a new tab in the active kitty window and split that tab into 6 windows, but I'm not sure what's the way to do that, if it's at all possible

3. Seems the titles I set don't show up on each of the split windows. Is that expected? Any way to make them show up so it's obvious what each window is? 

Thank you so much for the help!

**Comment by kovidgoyal** [comment, marked as answer]
1. zsh -ilc "exec whatever"
2. Use remote control
3. If you want separate titles for each window create OS Windows not
kitty windows.

**Reply by freak4pc** [reply]
Thank you so much! 

1 worked perfectly. 
2 I don't really know Remote Control but I'll investigate. If there is anything specific I can look at that'd be awesome
3 wouldn't that mean I just have many OS windows, and not a single window split into 6 inner-windows?

**Comment by kovidgoyal** [comment]
On Wed, Nov 20, 2024 at 11:53:45PM -0800, Shai Mishali wrote:
> Thank you so much!
>
> 1 worked perfectly.
> 2 I don't really know Remote Control but I'll investigate. If there is anything specific I can look at that'd be awesome

https://sw.kovidgoyal.net/kitty/remote-control/

> 3 wouldn't that mean I just have many OS windows, and not a single window split into 6 inner-windows?

Yes, and each would have its own titlebar with a title
